### PR TITLE
correct for-attribute, move secret-input down

### PR DESCRIPTION
**Pull request by websupporter**
Solves #99

### CODE REVIEW

**Comment by websupporter** [comment]
get rid of the input-field. its useless. add a filter instead.

**Comment by websupporter** [comment]
put two filter in the `get_secret`-methods instead. more useful.

File: antispam_bee.php [hunk 1]
```diff
@@ -2621,7 +2626,19 @@ public static function get_secret_for_post( $post_id ) {
...
+		$secret = substr( sha1( md5( 'comment-id' . self::$_salt . get_the_title( (int) $post_id ) ) ), 0, 10 );
+
+		/**
+		 * Filters the secret for a post, which is used in the textarea id attribute.
+		 *
+		 * @param string $secret The secret.
+		 * @param int    $post_id The post ID.
+		 */
+		return apply_filters(
+			'get_secret_id_for_post',
```
Comment (schlessera) [inline antispam_bee.php]: Prefix is missing here.

File: antispam_bee.php [hunk 2]
```diff
@@ -2609,7 +2601,20 @@ private static function _update_daily_stats()
...
+		$secret = substr( sha1( md5( self::$_salt . get_the_title( (int) $post_id ) ) ), 0, 10 );
+
+		/**
+		 * Filters the secret for a post, which is used in the textarea name attribute.
+		 *
+		 * @param string $secret The secret.
+		 * @param int    $post_id The post ID.
+		 */
+		return apply_filters(
+			'ab_get_secret_for_post',
```
Comment (schlessera) [inline antispam_bee.php]: Maybe add `name` in the filter?

**Comment by websupporter** [comment]
refactored `get_secret_for_post()` => `get_secret_name_for_post()`
added an `esc_attr()`
filter naming fixed

File: antispam_bee.php [hunk 3]
```diff
@@ -1101,13 +1101,13 @@ public static function precheck_incoming_request()
...
-		$hidden_field = self::get_key($_POST, 'comment');
-		$plugin_field = self::get_key( $_POST, self::get_secret_for_post( $post_id ) );
+		$hidden_field = self::get_key( $_POST, 'comment' );
+		$plugin_field = self::get_key( $_POST, self::get_secret_name_for_post( $post_id ) );
 
 		// Hidden field check
 		if ( empty($hidden_field) && ! empty($plugin_field) ) {
 			$_POST['comment'] = $plugin_field;
-			unset( $_POST[ self::get_secret_for_post( $post_id ) ] );
+			unset( $_POST[ self::get_secret_name_for_post( $post_id ) ] );
```
Comment (schlessera) [inline antispam_bee.php]: Shouldn't this be `unset( $plugin_field )`?

Comment (websupporter) [inline antispam_bee.php]: We could unset this too. But I think the general idea here is more get rid of the extra `$_POST`-field after the data has been transfered to `$_POST['comment']`

Comment (schlessera) [inline antispam_bee.php]: Ah, nevermind, I misread the code.

**Pull request merged**